subsys/bluetooth/host/conn.c

```diff
@@ -991,9 +1006,14 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 			 */
 			bt_conn_unref(conn);
 			break;
+		case BT_CONN_DISCONNECTED:
+			/* Can only happen if bt_disable is called whilst
+			 * in a connection.
+			 */
+			bt_conn_unref(conn);
+			break;
```

Comment:
In line 882-884 we have
```c
if (conn->state == state) {
	BT_WARN("no transition %s", state2str(state));
	return;
}
```

So the previous is correct: We cannot end up in this case

Reply:
Sorry, believe this was meant to go on the line below for `case BT_CONN_CONNECTED`.